solidity/contracts/BurntPixArchives.sol

```diff
@@ -134,23 +204,36 @@ contract BurntPixArchives is LSP8CappedSupply {
             }
             burntArchives[archiveId] = archive;
         }
+        // TODO: emit event so we can update UI
     }
 
     function mintArchive(bytes32 archiveId, address to) public {
         Archive memory archive = burntArchives[archiveId];
-        require(archive.creator == msg.sender, "BurntPixArchives: Only the archive creator can mint the archive");
+        // todo: do you need to require that the archive id exists?
+        require(
+            archive.creator == msg.sender,
+            "BurntPixArchives: Only the archive creator can mint the archive"
+        );
         _mint(to, archiveId, true, "");
+        // todo: should we emit an event here too?
     }
 
-    function getArchives(address contributor) public view returns (bytes32[] memory) {
+    function getArchives(
+        address contributor
+    ) public view returns (bytes32[] memory) {
         return contributions[contributor].archiveIds;
     }
 
     function getContributors() public view returns (address[] memory) {
         return contributors;
     }
 
-    function getContributions(address[] memory targetContributors) public view returns (uint256[] memory) {
+    // todo: This could be changed to 2 functions:
+    // getContributions by one address (already existing)
+    // getTop10Contributions with up address
```

Comment:
calculating getTop10Contributions on chain might be too costly because of sorting but i havent tried. you might be correct @Ledespana